### ISSUE TITLE
Enhancement: [lines-between-class-members]  Additional option to allow no lines between class property declarations

### ISSUE THREAD

**Issue by ffxsam** [typescript-eslint/typescript-eslint#5361]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/lines-between-class-members

### Description

There should be an option to exclude declaring types of class properties. With the rule enabled, classes can look rather ugly:

```ts
class Thing {
  private name: string;

  private size: number;

  private value: number;

 // ...
}
```

I'd suggest having another option, `exceptBetweenPropertyDeclarations` (unsure of the proper terminology here).

### Fail

```typescript
class Thing {
  private name: string;
  private size: number;
  private value: number;
  constructor(name: string, size: number, value: number) {
    this.name = name;
    this.size = size;
    this.value = value;
  }
}
```

### Pass

```typescript
class Thing {
  private name: string;
  private size: number;
  private value: number;

  constructor(name: string, size: number, value: number) {
    this.name = name;
    this.size = size;
    this.value = value;
  }
}
```

**Comment by bradzacher** [comment]
Thanks for the issue!

This is an extension rule. That means we take the base rule and extend it to support TS syntax or features.

We don't break from the base rule to add new features unless they are specifically related to TS.

In this instance - this request is not related to TS, thus we shall not implement it. If you'd like to see it, feel free to raise a request with the ESLint team.

**Comment by ffxsam** [comment]
@bradzacher Wouldn't this be TS-related since you can't do this in plain JS?

```ts
class Thing {
  private name: string;

  // ...
}
```
